### PR TITLE
Clean up threshold sign tests.

### PR DESCRIPTION
**Pull request by afck**
Add a test that actually verifies the signature, and refactor the tests a bit.

### CODE REVIEW

File: tests/threshold_sign.rs [hunk 1]
```diff
@@ -124,24 +156,20 @@ proptest! {
     })]
 
     #[test]
-    #[allow(clippy::unnecessary_operation)]
     fn test_threshold_sign_random_silent_200_samples(seed in gen_seed()) {
```
Comment (vkomenda) [inline tests/threshold_sign.rs]: This function should probably now be called `do_test_coin_random_silent_200_samples`.

File: tests/threshold_sign.rs [hunk 2]
```diff
@@ -124,24 +156,20 @@ proptest! {
...
-        do_test_threshold_sign_random_silent_200_samples(seed)
+        let new_adversary = || ReorderingAdversary::new();
+        test_coin_different_sizes(new_adversary, 200, seed);
     }
 
     #[test]
-    #[allow(clippy::unnecessary_operation)]
     fn test_threshold_sign_first_silent_50_samples(seed in gen_seed()) {
```
Comment (vkomenda) [inline tests/threshold_sign.rs]: This function should probably now be called `do_test_coin_random_silent_50_samples`.

File: tests/threshold_sign.rs [hunk 3]
```diff
@@ -13,30 +13,65 @@ use rand::{Rng, SeedableRng};
...
+        // Handle messages until all good nodes have terminated.
+        while !net.nodes().all(|node| node.algorithm().terminated()) {
+            let _ = net.crank_expect(&mut rng);
+        }
+        let node0 = net.correct_nodes().nth(0).unwrap();
+        // Verify that all instances output the same value.
+        assert_eq!(1, node0.outputs().len());
+        assert!(net.nodes().all(|node| node.outputs() == node0.outputs()));
+        let pk = node0.algorithm().netinfo().public_key_set().public_key();
+        assert!(pk.verify(&node0.outputs()[0], "message"));
```
Comment (vkomenda) [inline tests/threshold_sign.rs]: Maybe it's a minor detail but the expected value `"message"` could be cloned from the initial one to guarantee equality.

File: tests/threshold_sign.rs (hunk 1 above)
Comment (afck) [reply tests/threshold_sign.rs]: Done. (But without the `do_`.)

**Pull request merged**